### PR TITLE
Automated cherry pick of #90243: Revert "stop defaulting kubeconfig to http://localhost:8080"

### PR DESCRIPTION
**Pull request by soltysh**
Cherry pick of #90243 on release-1.18.

#90243: Revert "stop defaulting kubeconfig to http://localhost:8080"

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go [hunk 1]
```diff
@@ -124,6 +126,15 @@ func (o *Options) Run() error {
 		// Always request fresh data from the server
 		o.discoveryClient.Invalidate()
 		serverVersion, serverErr = o.discoveryClient.ServerVersion()
+		// FIXME soltysh:
+		// since we have defaulting to localhost:8080 in staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#getDefaultServer
+		// we need to hack around this when getting version and ignore reaching the localhost:8080 server
+		switch t := serverErr.(type) {
+		case *url.Error:
+			if strings.Contains(t.URL, "localhost:8080") {
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:134]: why is this needed? 1.17 would print an error if you did `kubectl version` with no config file

Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go]: Dropped.

**Comment by soltysh** [comment]
@kubernetes/patch-release-team for approval
